statsmodels/distributions/empirical_distribution.py

```diff
@@ -144,6 +144,48 @@ def __init__(self, x, side='right'):
         # which would have raised a NameError if hit, so would need to be
         # fixed.  See GH#5701.
 
+class EWCDF(StepFunction):
+    """
+    Return the Empirical Weighted CDF of an array as a step function.
+
+    Parameters
+    ----------
+    x : array_like
+        Observations
+    w : array_like
+        Weights of the observations
+    side : {'left', 'right'}, optional
+        Default is 'right'. Defines the shape of the intervals constituting the
+        steps. 'right' correspond to [a, b) intervals and 'left' to (a, b].
+
+    Returns
+    -------
+    Empirical Weighted CDF as a step function.
+
+    Examples
+    --------
+    >>>
+    >>> ewcdf = EWCDF([3, 3, 1, 4], [1, 1, 1, 1])  # For equal weights ewcdf result is same as ecdf (see above)
+    >>> ewcdf([3, 55, 0.5, 1.5])
+    array([0.75, 1.  , 0.  , 0.25])
+    >>>
+    >>> ewcdf = EWCDF([3, 3, 1, 4], [0.25, 1, 2.5, 5])
+    >>>
+    >>> ewcdf([3, 55, 0.5, 1.5])
+    array([0.42857143, 1., 0. , 0.28571429])
+    """
+    def __init__(self, x, w, side='right'):
+        assert len(w) == len(x)
+        w = np.array(w, copy=True)
```

Comment:
Why copy w?  Doesn't look like you change it.  Same goes for x I think.  Should add a test to verify that x and w aren't modified.